packages/opentelemetry-cloud-trace-propagator/test/CloudPropagator.test.ts

```diff
@@ -0,0 +1,186 @@
+
+import {
+  Context,
+  defaultSetter,
+  defaultGetter,
+  SpanContext,
+  TraceFlags,
+} from '@opentelemetry/api';
+import * as assert from 'assert';
+import {
+  setExtractedSpanContext,
+  getExtractedSpanContext,
+} from '@opentelemetry/core';
+import { CloudPropagator, X_CLOUD_TRACE_HEADER } from '../src/CloudPropagator';
+
+describe('CloudPropagator', () => {
+  const cloudPropagator = new CloudPropagator();
+  let carrier: { [key: string]: unknown };
+
+  beforeEach(() => {
+    carrier = {};
+  });
+
+  describe('.inject()', () => {
+    it('should inject a context of a sampled span', () => {
+      const spanContext: SpanContext = {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.SAMPLED,
+      };
+
+      cloudPropagator.inject(
+        setExtractedSpanContext(Context.ROOT_CONTEXT, spanContext),
+        carrier,
+        defaultSetter
+      );
+
+      assert.deepStrictEqual(
+        carrier[X_CLOUD_TRACE_HEADER],
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=1'
+      );
+    });
+    it('should inject a context of a unsampled span', () => {
+      const spanContext: SpanContext = {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.NONE,
+      };
+      cloudPropagator.inject(
+        setExtractedSpanContext(Context.ROOT_CONTEXT, spanContext),
+        carrier,
+        defaultSetter
+      );
+      assert.deepStrictEqual(
+        carrier[X_CLOUD_TRACE_HEADER],
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=0'
+      );
+    });
+  });
+
+  describe('.extract()', () => {
+    it('should extract context of a sampled span from carrier', () => {
+      carrier[X_CLOUD_TRACE_HEADER] =
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=1';
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.SAMPLED,
+        isRemote: true,
+      });
+    });
+
+    it('should extract context of a unsampled span from carrier', () => {
+      carrier[X_CLOUD_TRACE_HEADER] =
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=0';
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.NONE,
+        isRemote: true,
+      });
+    });
+
+    it('should handle invalid trace_flags', () => {
+      carrier[X_CLOUD_TRACE_HEADER] =
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=123';
```

Comment:
This is a valid flags value; it should result in the SAMPLED bit to be set since bit 0 is set in the flags.

Reply:
Makes sense, done.

Reply:
@kintel note that [this code](https://github.com/googleapis/google-cloud-dotnet/blob/master/apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderContext.cs#L44) from Google has a different interpretation.

I suggest to change the comment in the header description if this is changed.

Reply:
Yeah, that code should be updated, as that comment is confusing